FirebaseMessaging/Tests/UnitTests/FIRMessagingTokenOperationsTest.m

```diff
@@ -67,6 +69,39 @@ - (void)performTokenOperation {
 
 @end
 
+/// A fake heartbeat logger used for dependency injection during testing.
+@interface FIRHeartbeatLoggerFake : NSObject <FIRHeartbeatLoggerProtocol>
+@property(nonatomic, copy, nullable) FIRHeartbeatInfoCode (^onHeartbeatCodeForTodayHandler)(void);
+@end
+
+@implementation FIRHeartbeatLoggerFake
+
+- (nonnull FIRHeartbeatsPayload *)flushHeartbeatsIntoPayload {
+  // This API should not be used by the below tests because the Messaging
+  // SDK uses only the V1 heartbeat API (`heartbeatCodeForToday`) for
+  // getting a single heartbeat.
+  [self doesNotRecognizeSelector:_cmd];
+  return nil;
+}
```

Comment:
Just noticed this pattern across multiple SDKs - a potential improvement for the future would be to mark this and the `log` API as `@optional` in the Protocol (possible in ObjC but not in Swift). You'd need to check if the instance responds to it in the implementation but would reduce the amount of manual code you have to write and extra messaging.

Reply:
Thanks! Added a TODO to revisit (a3dc31a).